internal/controller/storageclient_controller.go

```diff
@@ -368,43 +372,84 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
-			err = r.reconcileResource(&csiopv1a1.CephConnection{}, kubeResource)
-		case csiopv1a1.GroupVersion.WithKind("ClientProfileMapping"):
-			err = r.reconcileResource(&csiopv1a1.ClientProfileMapping{}, kubeResource)
-		case corev1.SchemeGroupVersion.WithKind("Secret"):
-			err = r.reconcileResource(&corev1.Secret{}, kubeResource)
-		case nbv1.SchemeGroupVersion.WithKind("Noobaa"):
-			err = r.reconcileResource(&nbv1.NooBaa{}, kubeResource)
-		case storagev1.SchemeGroupVersion.WithKind("StorageClass"):
-			err = r.reconcileResource(&storagev1.StorageClass{}, kubeResource)
-		case snapapi.SchemeGroupVersion.WithKind("VolumeSnapshotClass"):
-			err = r.reconcileResource(&snapapi.VolumeSnapshotClass{}, kubeResource)
-		case groupsnapapi.SchemeGroupVersion.WithKind("VolumeGroupSnapshotClass"):
-			if val, _ := r.crdsBeingWatched.Load(VolumeGroupSnapshotClassCrdName); !val.(bool) {
-				continue
-			}
-			err = r.reconcileResource(&groupsnapapi.VolumeGroupSnapshotClass{}, kubeResource)
-		case replicationv1a1.GroupVersion.WithKind("VolumeReplicationClass"):
-			err = r.reconcileResource(&replicationv1a1.VolumeReplicationClass{}, kubeResource)
-		case csiopv1a1.GroupVersion.WithKind("ClientProfile"):
-			err = r.reconcileResource(&csiopv1a1.ClientProfile{}, kubeResource)
+	resourceMapping := gvkToRawresources{}
+	for idx := range storageClientResponse.KubeObjects {
+		resource := storageClientResponse.KubeObjects[idx]
+		if resource != nil && resource.GroupVersionKind != "" {
+			resourceMapping[resource.GroupVersionKind] = append(resourceMapping[resource.GroupVersionKind], resource.Bytes)
 		}
-		if err != nil {
-			return reconcile.Result{}, err
+	}
+	if err := r.reconcileResourceByGVK(
+		quotav1.SchemeGroupVersion.WithKind("ClusterResourceQuota"),
+		&quotav1.ClusterResourceQuota{},
+		resourceMapping,
+	); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to reconcile %s: %v", "ClusterResourceQuota", err)
+	}
+	if err := r.reconcileResourceByGVK(
+		csiopv1a1.GroupVersion.WithKind("CephConnection"),
+		&csiopv1a1.CephConnection{},
+		resourceMapping,
+	); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to reconcile %s: %v", "CephConnection", err)
+	}
+	if err := r.reconcileResourceByGVK(
+		csiopv1a1.GroupVersion.WithKind("ClientProfileMapping"),
+		&csiopv1a1.ClientProfileMapping{},
+		resourceMapping,
+	); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to reconcile %s: %v", "ClientProfileMapping", err)
+	}
+	if err := r.reconcileResourceByGVK(
+		corev1.SchemeGroupVersion.WithKind("Secret"),
+		&corev1.Secret{},
+		resourceMapping,
+	); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to reconcile %s: %v", "Secret", err)
+	}
+	if err := r.reconcileResourceByGVK(
+		nbv1.SchemeGroupVersion.WithKind("NooBaa"),
+		&nbv1.NooBaa{},
+		resourceMapping,
+	); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to reconcile %s: %v", "Noobaa", err)
+	}
```

Comment:
There is no real reason to have these cases hardcoded one after another. We do want the code to dictate which types we should be processing, but we could have it in the form of a master fixed list of just client.Object instances where the type info is already filled in.